joblib/test/test_hashing.py

```diff
@@ -398,6 +398,15 @@ def test_hashes_stay_the_same_with_numpy_objects():
     # 'f8' for float arrays because 'f8' arrays generated by
     # rng.random.randn don't seem to be bit-identical on 32bit and
     # 64bit machines.
+
+    lambda_func = lambda: 1
+    if PY3:
+        if sys.version[:3] in ('3.4', '3.5'):
+            lambda_func.__code__ = compile('1', '<string>', 'exec')
```

Comment:
`__code__` stores the path of the lambda function's file so this gets rid of it as testing environments can have different paths. It's not the ideal solution,  but I couldn't come up with anything better. I think properly done dill should accept a path option so lambdas can store relative paths in `__code__`.